src/wp-admin/options.php

```diff
@@ -195,18 +195,28 @@
 	 * @param bool $enabled Whether post-by-email configuration is enabled. Default true.
 	 */
 	if ( apply_filters( 'enable_post_by_email_configuration', true ) ) {
-		$whitelist_options['writing'] = array_merge( $whitelist_options['writing'], $mail_options );
+		$allowed_options['writing'] = array_merge( $allowed_options['writing'], $mail_options );
 	}
 }
 
 /**
- * Filters the options whitelist.
+ * Filters the allowed options list.
  *
  * @since 2.7.0
+ * @deprecated 5.5.0 Use {@see 'allowed_options'} instead.
  *
- * @param array $whitelist_options The options whitelist.
+ * @param array $allowed_options The allowed options list.
  */
-$whitelist_options = apply_filters( 'whitelist_options', $whitelist_options );
+$allowed_options = apply_filters_deprecated( 'whitelist_options', array( $allowed_options ), '5.5.0', 'apply_filters_deprecated', __( 'Please join us in writing more inclusive code.' ) );
+
```

Comment:
Maybe we add something like?

```php
if ( ! empty( $whitelist_options ) ) {
    _doing_it_wrong();
    $allowed_options = array_merge( $allowed_options, $whitelist_options );
}
```

Reply:
What would the benefit be for this vs. calling `apply_filters_deprecated()`?

Reply:
This wouldn't be an instead of, but an in addition to if people manually accessed the `$allowed_options` global. Though I may have misunderstood if that was a concern.

Reply:
I don't believe that can happen in this scenario. `$allowed_options` is local only to this file and it gets defined on line 83. I think `$new_whitelist_options` (which is a `global` variable) may be what you are thinking of? I opened a [new Trac ticket to tackle that](https://core.trac.wordpress.org/ticket/50434).